### ISSUE TITLE
Improve code quality by using standard tools to format and analyse for PRs

### ISSUE THREAD

**Issue by thomasmerz** [speed47/spectre-meltdown-checker#429]
Your `spectre-meltdown-checker.sh` is already `shellcheck`'ed, but you don't have a **GitHub Action** to enforce this before a Pull Requests can be merged.

And I have seen, that you are doing some "low-level" / very simple format-checking. I suggest to use `shfmt` which formats shell programs. This can also be enforced by a **GitHub Action** before a Pull Request can be merged.

If you @speed47 agree I will make a PR with my suggestion regarding a nice GitHub Action for every push on this script and on every PR.

**Comment by speed47** [comment]
Yes, that would be a good idea! You'll probably have to find out which `shfmt` parameters work best to avoid rewriting the entire script (I don't want to lose all the `git blame` usefulness), and mainly ensure that future PR just respect the current coding style.

That's also why I always left tabs (I usually use spaces!), because I noticed that too late and didn't want to rewrite the entire git history :)

**Comment by thomasmerz** [comment]
Challenge accepted 👍🏻
This might take some time, because it's not in a hurry and I'll be looking for some "creative time"…

**Comment by thomasmerz** [comment]
@speed47 , can you please review my PR #435 and merge if you are satisfied? Thanks a lot 👍

**Comment by thomasmerz** [comment]
Ping @speed47

**Comment by speed47** [comment]
Hey @thomasmerz , sorry about that, my maintainership had gone dormant for a few years due to various personal reasons, and now that I have some time to update this script, I also took the opportunity to completely split the script in snippets, separating "source" from "release" branches, as explained here: https://github.com/speed47/spectre-meltdown-checker/releases/tag/v26.21.0401891 and shellcheck/shfmt have been added along the way.

Thanks for the time you took to make this project better, in any case!
